src/GafferScene/CollectScenes.cpp

```diff
@@ -80,24 +204,46 @@ class CollectScenes::SourcePathScope : public SourceScope
 		SourcePathScope( const Context *context, const CollectScenes *collectScenes, const ScenePlug::ScenePath &downstreamPath )
 			:	SourceScope( context, collectScenes->rootNameVariablePlug()->getValue() )
 		{
-			setRootName( downstreamPath[0] );
-			ScenePlug::ScenePath upstreamPath;
-			// We evaluate the sourceRootPlug _after_ setting the root name,
-			// so that users can use the root name in expressions and
-			// substitutions.
-			ScenePlug::stringToPath( collectScenes->sourceRootPlug()->getValue(), upstreamPath );
-			upstreamPath.insert( upstreamPath.end(), downstreamPath.begin() + 1, downstreamPath.end() );
-			set( ScenePlug::scenePathContextName, upstreamPath );
+			// Evaluate RootTree in global scope.
+			remove( ScenePlug::scenePathContextName );
+			m_rootTree = boost::static_pointer_cast<const RootTree>( collectScenes->rootTreePlug()->getValue() );
```

Comment:
Sorry for the naive q, but what is the motivation for storing `m_rootTree` instead of a suitable ref pointer to the location?

Reply:
We could do that, but we'd either need to derive Location from `RefCounted`, or store `shared_ptr` in `Location::ChildMap`. Both have greater overhead than the current approach.